File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.cpp
#include <algorithm>
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, vector<int>> counter;
        vector<int> seen;
        vector<int> result;

        for (int& num: nums) {
            if (find(seen.begin(), seen.end(), num) != seen.end()) continue;
            else {
                int count = std::count(nums.begin(), nums.end(), num);
                counter[count].push_back(num);
                seen.push_back(num);
            }
        }
        
        for (map<int, vector<int>>::reverse_iterator it = counter.rbegin(); it != counter.rend(); ++it) {
            for (int num: it->second) {
                if (k == 0) return result;
                result.push_back(num);
                k -= 1;
            }
        }
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-8.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        prefix = nums[0]

        result = [1] * len(nums)

        for i in range(1, len(nums)):
            result[i] = prefix
            prefix *= nums[i]

        suffix = nums[-1]

        for i in range(len(nums) - 2, -1, -1):
            result[i] *= suffix
            suffix *= nums[i]
        
        return result


        